mne/inverse_sparse/mxne_optim.py

```diff
@@ -220,34 +220,31 @@ def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
                 for k in range(K):
                     U[k] = last_K_X[k + 1].ravel() - last_K_X[k].ravel()
                 C = U @ U.T
-                one_vec = np.ones(K)
-
-                try:
-                    z = np.linalg.solve(C, one_vec)
-                except np.linalg.LinAlgError:
-                    # Matrix C is not always expected to be non-singular. If C
-                    # is singular, acceleration is not used at this iteration
-                    # and the solver proceeds with the non-sped-up code.
+                # at least on ARM64 we can't rely on np.linalg.solve to
+                # reliably raise LinAlgError here, so use SVD instead
+                u, s, _ = np.linalg.svd(C, hermitian=True)
+                if s[-1] <= 1e-6 * s[0]:
                     logger.debug("Iteration %d: LinAlg Error" % (i + 1))
-                else:
-                    c = z / z.sum()
-                    X_acc = np.sum(
-                        last_K_X[:-1] * c[:, None, None], axis=0
+                    continue
+                z = ((u * 1 / s) @ u.T).sum(0)
```

Comment:
@larsoner can you put a comment that this solves `z = np.linalg.solve(C, one_vec)`
it took me a few seconds to understand why